### PR TITLE
Run load tests on Kibana CI

### LINKED ISSUE #7: Enable monitoring to Kibana stats
To have more information about Kibana & ES performance during scenario run, we want to specify Kibana stats instance as monitoring for each new deployment

### PR DESCRIPTION
**Pull request by dmlemeshko**
Closes #7
This PR allows running load testing on Kibana CI:

- Elastic Staging Cloud (to include snapshots builds) 
- tests are limited to DemoJourney for now
- its own Jenkins [job](https://kibana-ci.elastic.co/job/elastic+kibana+load-testing/)
- Ingest results into Kibana-stats

image

Tested on 7.9.3, 7.10.0 and 7.11.0-SNAPSHOT

### CODE REVIEW

File: src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala [hunk 1]
```diff
@@ -13,7 +13,8 @@ import spray.json.lenses.JsonLenses._
 import spray.json.DefaultJsonProtocol._
```
Comment (wayneseymour) [inline src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala:13]: Curious, what's the underscore mean here?

File: src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala [hunk 2]
```diff
@@ -26,21 +27,21 @@ class CloudHttpClient {
...
       s"preparePayload: Using ${deployPayloadTemplate} payload template"
     )
     val template = Helper.loadJsonString(deployPayloadTemplate)
-    logger.info(s"preparePayload: Using ${config.toString}")
+    logger.info(
+      s"preparePayload: Stack version ${stackVersion} with ${config.toString} configuration"
+    )
 
     val payload = template
       .update('name ! set[String](s"load-testing-${Instant.now.toEpochMilli}"))
```
Comment (wayneseymour) [inline src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala:40]: Is this string concatenation?  Perhaps it's adding a string to set of strings?

File: src/test/scala/org/kibanaLoadTest/helpers/HttpHelper.scala [hunk 3]
```diff
@@ -27,43 +30,92 @@ class HttpHelper(appConfig: KibanaConfiguration) {
...
+      responseBody = EntityUtils.toString(statusResponse.getEntity, "UTF-8")
+    } catch {
+      case _: Throwable => {
+        logger.error("Exception occurred during getting Kibana status")
+      }
+    } finally {
+      httpClient.close()
+    }
+
+    responseBody
```
Comment (wayneseymour) [inline src/test/scala/org/kibanaLoadTest/helpers/HttpHelper.scala:119]: Last line return?  That's what it seems like.

File: src/test/scala/org/kibanaLoadTest/ingest/Main.scala [hunk 4]
```diff
@@ -9,7 +9,9 @@ import org.kibanaLoadTest.helpers.Helper.{getLastReportPath, getTargetPath}
 
 object Main {
   def main(args: Array[String]): Unit = {
-    val host = System.getenv("HOST_FROM_VAULT")
+    val hostValue = System.getenv("HOST_FROM_VAULT")
+    val host =
+      if (hostValue.startsWith("http")) hostValue else "https://" + hostValue
```
Comment (wayneseymour) [inline src/test/scala/org/kibanaLoadTest/ingest/Main.scala:14]: Oh I love this, `if` as an expression, not a statement.

File: src/test/scala/org/kibanaLoadTest/simulation/BaseSimulation.scala [hunk 5]
```diff
@@ -35,19 +38,23 @@ class BaseSimulation extends Simulation {
...
+      logger.info(s"Getting Kibana status info")
+      val response = new HttpHelper(appConfig).getStatus()
       val meta = Map(
         "deploymentId" -> appConfig.deploymentId.get,
         "baseUrl" -> appConfig.baseUrl,
-        "version" -> appConfig.buildVersion
+        "version" -> appConfig.buildVersion,
+        "buildHash" -> response.extract[String]('version / 'build_hash),
+        "buildNumber" -> response
+          .extract[Int]('version / 'build_number)
```
Comment (wayneseymour) [inline src/test/scala/org/kibanaLoadTest/simulation/BaseSimulation.scala:49]: This `[Int]` is a type-cast?

File: src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala (hunk 1 above)
Comment (wayneseymour) [reply src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala:13]: I've seen it mean, "ignore this"...like a placeholder of something you dont care about.

**Review by wayneseymour — APPROVED** [review]
LGTM, but I've some simple questions

File: src/test/scala/org/kibanaLoadTest/helpers/HttpHelper.scala (hunk 3 above)
Comment (dmlemeshko) [reply src/test/scala/org/kibanaLoadTest/helpers/HttpHelper.scala:119]: exactly. In scala there is no need to use `return` statement. Last line result is returned unless function output is defined as "Unit"

File: src/test/scala/org/kibanaLoadTest/ingest/Main.scala (hunk 4 above)
Comment (dmlemeshko) [reply src/test/scala/org/kibanaLoadTest/ingest/Main.scala:14]: no ternary operator :)

File: src/test/scala/org/kibanaLoadTest/simulation/BaseSimulation.scala (hunk 5 above)
Comment (dmlemeshko) [reply src/test/scala/org/kibanaLoadTest/simulation/BaseSimulation.scala:49]: yes, when you read json path with lenses you need to explicitly set value type

File: src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala (hunk 2 above)
Comment (dmlemeshko) [reply src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala:40]: `s"load-testing-${Instant.now.toEpochMilli}"` is string concatenation, but the whole expression updates json string parameter `name` with a new string value

File: src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala (hunk 1 above)
Comment (dmlemeshko) [reply src/test/scala/org/kibanaLoadTest/helpers/CloudHttpClient.scala:13]: Wild card: all of `spray.json.DefaultJsonProtocol` is imported

**Pull request merged**